backend/zeno_backend/database/select.py

```diff
@@ -51,115 +110,96 @@ def models(project: str) -> list[str]:
     return [m[0] for m in model_results]
 
 
-def projects(user: User) -> list[Project]:
+def projects(user: User, projects_request: ProjectsRequest) -> list[Project]:
     """Get all projects available to the user.
 
     Args:
         user (User): the user for which to fetch the available projects.
+        projects_request (ProjectsRequest): the request object.
 
     Returns:
         list[Project]: the projects that the user can interact with.
     """
     with Database() as db:
-        own_projects_result = db.execute_return(
-            "SELECT uuid, name, view, "
-            "samples_per_page, public, description FROM projects WHERE owner_id = %s;",
-            [user.id],
-        )
-        own_projects = list(
-            map(
-                lambda project: Project(
-                    uuid=project[0],
-                    name=project[1],
-                    owner_name=user.name,
-                    view=project[2],
-                    samples_per_page=project[3],
-                    editor=True,
-                    public=project[4],
-                    description=project[5],
-                ),
-                own_projects_result,
+        params = [user.id, user.id, user.id]
+        likes_query = sql.SQL(
+            "SELECT project_uuid, COUNT(*) as total_likes"
+            " FROM project_like GROUP BY project_uuid "
+        )
+        projects_query = (
+            sql.SQL("WITH LikesSummary AS (")
+            + likes_query
+            + sql.SQL("),")
+            + sql.SQL(" CombinedProjects AS (")
+            + PROJECTS_BASE_QUERY
+            + sql.SQL(") ")
+            + sql.SQL(
+                "SELECT cp.*, COALESCE(ls.total_likes, 0) AS total_likes FROM"
+                " CombinedProjects cp LEFT JOIN LikesSummary ls "
+                " ON cp.uuid = ls.project_uuid ORDER BY total_likes DESC "
```

Comment:
This is very helpful.

Reply:
Something to be considered: What if we want to sort by something different than total number of likes?

Reply:
Yeah I think future PR will make this an option and use different types of sorts